### PR TITLE
Solve 3

### PR DESCRIPTION
**Pull request by qarux**

### CODE REVIEW

File: 3_ecosystem/src/db.rs [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+  pub name : String,
+  pub password_hash : String,
+}
+
+pub async fn init_db() -> SqlitePool
+{
+  let config = SqliteConnectOptions::from_str( ":memory:" ).unwrap();
+  let pool = SqlitePool::connect_with( config ).await.unwrap();
+  sqlx::migrate!( "./migrations" )
+    .run( &pool )
```
Comment (dmvict) [inline 3_ecosystem/src/db.rs:18]: Please, update formatting.

File: 3_ecosystem/src/main.rs [hunk 2]
```diff
@@ -1,4 +1,68 @@
...
+mod db;
+mod graphql;
+
+#[ actix_web::main ]
+async fn main() -> std::io::Result< () >
+{
+  let db_pool = init_db().await;
+  let schema = web::Data::new( create_schema( db_pool ) );
+
+  HttpServer::new( move ||
```
Comment (dmvict) [inline 3_ecosystem/src/main.rs:39]: Please, update formatting.

**Comment by dmvict** [comment]
Can be merged.

File: 3_ecosystem/src/db.rs (hunk 1 above)
Comment (qarux) [reply 3_ecosystem/src/db.rs]: fixed

File: 3_ecosystem/src/main.rs (hunk 2 above)
Comment (qarux) [reply 3_ecosystem/src/main.rs]: fixed

**Pull request merged**